raw.go

```diff
@@ -31,3 +31,32 @@ type RawSpan struct {
 	// The span's associated baggage.
 	Baggage map[string]string // initialized on first use
 }
+
+// SpanDetail summarizes the identifying fields of a Span, for
+// diagnostic and implementation-specific purposes.
+type SpanDetail struct {
```

Comment:
@jmacd made the good point this morning that – since the impl objects are immediately recycled by the pool – any access provided via a type conversion to something akin to the `BasicSpan` interface above would have to come along with strong caveats about calling any methods prior to `Finish()`.

Anyway, I still like the interface approach.